web/components/feed/feed-comments.tsx

```diff
@@ -1,35 +1,487 @@
+              </>
+            )}
+          </>
+          <CopyLinkDateTimeComponent
+            contract={contract}
+            createdTime={createdTime}
+            elementId={comment.id}
+          />
+        </p>
+        <TruncatedComment
+          comment={text}
+          moreHref={contractPath(contract)}
+          shouldTruncate={truncate}
+        />
+        {onReplyClick && (
+          <button
+            className={'text-xs font-bold text-gray-500 hover:underline'}
+            onClick={() => onReplyClick(comment)}
+          >
+            Reply
+          </button>
+        )}
+      </div>
+    </Row>
+  )
+}
 
-// TODO: move feed commment and comment thread in here when sinclair confirms they're not working on them rn
 export function getMostRecentCommentableBet(
   betsByCurrentUser: Bet[],
-  comments: Comment[],
+  commentsByCurrentUser: Comment[],
   user?: User | null,
   answerOutcome?: string
 ) {
   return betsByCurrentUser
     .filter((bet) => {
       if (
         canCommentOnBet(bet, user) &&
-        // The bet doesn't already have a comment
-        !comments.some((comment) => comment.betId == bet.id)
+        !commentsByCurrentUser.some(
+          (comment) => comment.createdTime > bet.createdTime
+        )
       ) {
         if (!answerOutcome) return true
         // If we're in free response, don't allow commenting on ante bet
-        return (
-          bet.outcome !== GENERAL_COMMENTS_OUTCOME_ID &&
-          answerOutcome === bet.outcome
-        )
+        return answerOutcome === bet.outcome
       }
       return false
     })
     .sort((b1, b2) => b1.createdTime - b2.createdTime)
     .pop()
 }
 
+export function CommentInput(props: {
+  contract: Contract
+  betsByCurrentUser: Bet[]
+  commentsByCurrentUser: Comment[]
+  // Tie a comment to an free response answer outcome
+  answerOutcome?: string
+  // Tie a comment to another comment
+  parentComment?: Comment
+  replyToUsername?: string
+  setRef?: (ref: HTMLTextAreaElement) => void
+}) {
+  const {
+    contract,
+    betsByCurrentUser,
+    commentsByCurrentUser,
+    answerOutcome,
+    parentComment,
+    replyToUsername,
+    setRef,
+  } = props
+  const user = useUser()
+  const [comment, setComment] = useState('')
+  const [focused, setFocused] = useState(false)
+
+  const mostRecentCommentableBet = getMostRecentCommentableBet(
+    betsByCurrentUser,
+    commentsByCurrentUser,
+    user,
+    answerOutcome
+  )
+  const { id } = mostRecentCommentableBet || { id: undefined }
+
+  useEffect(() => {
+    if (!replyToUsername || !user || replyToUsername === user.username) return
+    const replacement = `@${replyToUsername} `
+    setComment(replacement + comment.replace(replacement, ''))
+  }, [user, replyToUsername])
+
+  async function submitComment(betId: string | undefined) {
+    if (!user) {
+      return await firebaseLogin()
+    }
+    if (!comment) return
+
+    // Update state asap to avoid double submission.
```

Comment:
Another solution FYI would be to have an isSubmitting boolean, and call setIsSubmitting before and after the async call. Then disable submits while isSubmitting is true.

This solution works ok though, unless we need to handle errors or want the visual feedback of showing the comment button disabled on submit.